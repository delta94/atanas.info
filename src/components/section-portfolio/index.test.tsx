import * as React from 'react';
import { shallow } from 'enzyme';

import { SectionPortfolio } from '.';

jest.mock('../../assets/scripts/projects-list.json', () => ({
	default: [
		{
			url: 'https://www.google.com',
			image: 'image 1',
			title: 'Project 1',
			description: 'Description 1'
		},
		{
			url: '',
			image: 'image 2',
			title: 'Project 2',
			description: 'Description 2'
		}
	]
}));

describe('SectionPortfolio component', () => {
	it('Should render the SectionPortfolio component', () => {
		const wrapper = shallow(<SectionPortfolio />);

		expect(wrapper).toMatchSnapshot();
	});
});
