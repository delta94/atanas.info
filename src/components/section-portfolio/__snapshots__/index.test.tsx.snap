// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionPortfolio component Should render the SectionPortfolio component 1`] = `
<Component
  actions={
    <Unknown
      onClick={[Function]}
    >
      Show all
    </Unknown>
  }
  hasButton={true}
  hasShell={true}
  id="portfolio"
>
  <h1>
    Portfolio
  </h1>
  <h2>
    Mobile applications
  </h2>
  <div
    className="c-section__sliders"
  >
    <div
      className="c-section__slider"
      key="0"
    >
      <Component
        href="https://apps.apple.com/gb/app/e-on-home-solar-and-heating/id1468486784"
      >
        <h3>
          E.ON Home for iPhone
        </h3>
        <p>
          Smart energy management application for iOS
        </p>
        <code>
          TypeScript, SCSS, React Native, Redux, MS Azure, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="E.ON Home for iPhone screenshot 1"
            src="images/temp/eon-home-iphone-dark1.png"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="E.ON Home for iPhone screenshot 2"
            src="images/temp/eon-home-iphone-dark2.png"
          />
        </div>
        <div
          key="2"
        >
          <img
            alt="E.ON Home for iPhone screenshot 3"
            src="images/temp/eon-home-iphone-dark3.png"
          />
        </div>
        <div
          key="3"
        >
          <img
            alt="E.ON Home for iPhone screenshot 4"
            src="images/temp/eon-home-iphone-dark4.png"
          />
        </div>
        <div
          key="4"
        >
          <img
            alt="E.ON Home for iPhone screenshot 5"
            src="images/temp/eon-home-iphone-light1.png"
          />
        </div>
        <div
          key="5"
        >
          <img
            alt="E.ON Home for iPhone screenshot 6"
            src="images/temp/eon-home-iphone-light2.png"
          />
        </div>
        <div
          key="6"
        >
          <img
            alt="E.ON Home for iPhone screenshot 7"
            src="images/temp/eon-home-iphone-light3.png"
          />
        </div>
        <div
          key="7"
        >
          <img
            alt="E.ON Home for iPhone screenshot 8"
            src="images/temp/eon-home-iphone-light4.png"
          />
        </div>
      </Slider>
    </div>
    <div
      className="c-section__slider"
      key="1"
    >
      <Component
        href="https://play.google.com/store/apps/details?id=com.eon.home.eu&hl=en"
      >
        <h3>
          E.ON Home for Android
        </h3>
        <p>
          Smart energy management application for Android
        </p>
        <code>
          TypeScript, SCSS, React Native, Redux, MS Azure, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="E.ON Home for Android screenshot 1"
            src="images/temp/eon-home-android1.webp"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="E.ON Home for Android screenshot 2"
            src="images/temp/eon-home-android2.webp"
          />
        </div>
        <div
          key="2"
        >
          <img
            alt="E.ON Home for Android screenshot 3"
            src="images/temp/eon-home-android3.webp"
          />
        </div>
        <div
          key="3"
        >
          <img
            alt="E.ON Home for Android screenshot 4"
            src="images/temp/eon-home-android4.webp"
          />
        </div>
      </Slider>
    </div>
    <div
      className="c-section__slider"
      key="2"
    >
      <Component
        href="https://apps.apple.com/gb/app/e-on-home-solar-and-heating/id1468486784#?platform=ipad"
      >
        <h3>
          E.ON Home for iPad
        </h3>
        <p>
          Smart energy management application for iOS
        </p>
        <code>
          TypeScript, SCSS, React Native, Redux, MS Azure, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="E.ON Home for iPad screenshot 1"
            src="images/temp/eon-home-ipad-dark.jpg"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="E.ON Home for iPad screenshot 2"
            src="images/temp/eon-home-ipad-light.jpg"
          />
        </div>
      </Slider>
    </div>
  </div>
  <h2>
    Web applications
  </h2>
  <div
    className="c-section__body"
  >
    <Component
      className=""
      href="https://www.google.com"
      key="0"
    >
      <div>
        <strong>
          Project 1
        </strong>
        <span>
          <small>
            Technologies:
          </small>
          Description 1
        </span>
      </div>
      <object
        data="image 1"
        title="Project 1"
        type="image/png"
        width="600"
      >
        <img
          loading="lazy"
          width="600"
        />
      </object>
    </Component>
    <Component
      className="disabled"
      href=""
      key="1"
    >
      <div>
        <strong>
          Project 2
        </strong>
        <span>
          <small>
            Technologies:
          </small>
          Description 2
        </span>
      </div>
      <object
        data="image 2"
        title="Project 2"
        type="image/png"
        width="600"
      >
        <img
          loading="lazy"
          width="600"
        />
      </object>
    </Component>
  </div>
</Component>
`;
